math/gklib/Portfile

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           cmake 1.1
+PortGroup           github 1.0
+
+github.setup        KarypisLab GKlib 08b72b2c41f0ac2a825438649ee7361bf0b488c3
+name                gklib
+version             20221128
+categories          math
+platforms           darwin
+maintainers         {@catap korins.ky:kirill} openmaintainer
+license             Apache-2
+
+description         KarypisLab library of various helper routines and frameworks used by the lab's software
+long_description    {*}${description}
+
+checksums           rmd160  a6ad0814ebf3cc0c39bdd68601c95cadbc37499d \
+                    sha256  66e6881fe1568b703e75c625f80de9707e23e1fa2b72e8501914516295b7626b \
+                    size    186076
+
+compiler.thread_local_storage   yes
+
+configure.args-append \
+                    -DBUILD_SHARED_LIBS=ON
+
+if {${configure.build_arch} ni [list i386 x86_64]} {
+    configure.args-append \
+                    -DNO_X86=ON
+}
```

Comment:
`build_arch` is only relevant for non-universal builds, so this stanza will not do the right thing when the +universal variant is used.

What is the right thing? What do you want to happen if the user requests a universal build for example for arm64 and x86_64...
 * ...when the build is performed on an Intel machine (where `build_arch` will be x86_64)?
 * ...when the build is performed on an Apple Silicon machine (where `build_arch` will be arm64)?

Reply:
@ryandesign this option is used to disable / enable this pice of code: https://github.com/KarypisLab/GKlib/blob/master/test/gkuniq.c#L69-L88

So, I need: add this option only on ppc and arm64 builds.

Reply:
does this addresses @ryandesign his review - I am not sure.... 

My question would be then - why use the inverse logic with "ni" and just make clear when you want to add this argument?

Reply:
@reneeotten an issue that this argument is disabling optimization (asm code) for x86_64 / i386 => and it explains the logic with `ni`: I'd like to disable it on unsupported architectures.

Reply:
I understand that - I am just of the opinion that listing the architectures where you want this option in is more clear than using the reverse logic of listing the ones where you don't want the option on.